### PR TITLE
Add Component

### PR DESCRIPTION
**Pull request by hangingchads**
Add initial iteration of rich-text-editor component

### CODE REVIEW

File: src/sh-rich-text-editor.scss [hunk 1]
```diff
@@ -0,0 +1,1004 @@
+@import '../node_modules/sh-core/src/base';
+
+#react-quill-editor {
+    display: block;
+    float: left;
+    width: 100%;
+
+    .quill-toolbar.ql-toolbar {
+        background: white;
```
Comment (SuperheroUI) [inline src/sh-rich-text-editor.scss]: use sass variable

File: src/sh-rich-text-editor.scss [hunk 2]
```diff
@@ -0,0 +1,1004 @@
...
+        margin-bottom: 10px;
+        border-radius: 2px;
```
Comment (SuperheroUI) [inline src/sh-rich-text-editor.scss]: use sass from base

File: src/sh-rich-text-editor.scss [hunk 3]
```diff
@@ -0,0 +1,1004 @@
...
+    .ql-snow.ql-toolbar .ql-picker.ql-font {
+        width: 160px;
+    }
+
+    .ql-snow.ql-toolbar .ql-picker.ql-size {
+        width: 90px;
+    }
+
+    .quill-contents-label {
+        background: white;
```
Comment (SuperheroUI) [inline src/sh-rich-text-editor.scss]: use sass on all colors, fonts size fonts, border radiuses, font families yadda yadda.

**Comment by hangingchads** [comment]
I made the requested changes and I pushed my changes.

File: src/quill.snow.scss [hunk 4]
```diff
@@ -0,0 +1,919 @@
+
```
Comment (kevinlbatchelor) [inline src/quill.snow.scss]: can we write this like sass and we should set up a sass linter that check for nesting and combining duplicate selectors and such.

File: example/App.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import ReactDOM from 'react-dom';
+import ShForm from 'sh-form';
+import ShRichTextEditor from '../bin/sh-rich-text-editor';
+require('../node_modules/sh-core/bin/main.css');
```
Comment (kevinlbatchelor) [inline example/App.js]: this should be an import

**Comment by hangingchads** [comment]
I made the requested changes.  Please note that this will require the new version of sh-core

**Comment by hangingchads** [comment]
I made the UI fixes.

File: src/sh-rich-text-editor.js [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+            }
+        });
+        this.props.onChange(value);
+    };
+
+    handleChangeSelection(range, oldRange, source) {
+        this.props.onChangeSelection(range, oldRange, source);
+    };
+
+    handleFocus(event) {
```
Comment (jhaddow) [inline src/sh-rich-text-editor.js]: I concerned that we removed the methods to focus on the editor. The point of those methods was not to handle a focus event, but to actually focus the text cursor on the editor. Do you have any thoughts on this @kevinlbatchelor?

**Comment by hangingchads** [comment]
I made the additional changes

**Comment by hangingchads** [comment]
I made some minor style fixes on Wednesday.

**Comment by hangingchads** [comment]
I made the changes that you requested.

**Pull request merged**